src/project.coffee

```diff
@@ -39,6 +39,14 @@ class Project extends Model
     @emitter = new Emitter
     @buffers ?= []
 
+    @directoryProviders = [new DefaultDirectoryProvider()]
+    atom.packages.serviceHub.consume(
+      'atom.directory-provider',
+      '^0.1.0',
+      # New providers are added to the front of @directoryProviders because
+      # DefaultDirectoryProvider is a catch-all that will always provide a Directory.
+      (provider) => @directoryProviders.splice(0, 0, provider))
```

Comment:
@bolinfest You're largely responsible for pushing me toward avoiding the requirement of running code to make these kinds of decisions. What do you think?